stories/cli/steps.fmf

```diff
@@ -163,3 +163,39 @@
         - tmt run finish
     link:
       - implemented-by: /tmt/steps/finish
+
+/allowed-how:
+    story: Allow only a subset of possible step plugins to be used
+    description: |
+        I wish to limit tmt to only a subset of plugins available
+        for a particular step. This can be achieved by manipulating
+        installed plugins, moving them around to different directories,
+        but that would be impractical.
+
+        Instead, every step subcommand offers ``--allowed-how`` option
+        which limits methods allowed to be used in the step. The option
+        accepts a regular expression allowed methods - names set with
+        ``how`` key or ``--how`` option - must match to be allowed.
```

Comment:
@happz could we mention `re.match` is used for matching, that should make the matching behaviour more clear to the user.

Reply:
I guess we could mention it, and we should, although I'd rather see this in a form of a more generic page or section because this rule is not specific to this option alone. Something like "whenever tmt evaluates regular expression provided via fmf key or CLI option, Python's `re.match` is used". This applies to `plan/test --name` or `tmt * lint ...`, and we should have a single point here "tmt & regex" is explained, and other pages could point to that paragraph. I would gladly add it in an extra PR rather than adding it here and now.

Reply:
Filed https://github.com/teemtee/tmt/issues/2602

Reply:
@thrix ^^ would this be acceptable? If so, we could merge this PR right away :)

Reply:
Yeah, that sounds good to me as well.